### PR TITLE
Restructure Asymmetric Algorithm Chapters in Cryptodoc.

### PR DESCRIPTION
**Pull request by reneme**
Operations on asymmetric algorithms are now organized per-algoritm. I.e. the top-level structure is RSA, DSA, Kyber, ... and keygen, sign, verify, KEX, ... are sub-sections of the respective algos.

I carefully made sure not to loose any content-pieces and didn't change the wording significantly.

The new document structure looks like below, with each algorithm having sub-chapters about param-gen, key-gen, sig, enc, verify, KEM, KEX, ...

* [...]
* Public Key Algorithms
  * Diffie-Hellman
  * DSA
  * Elliptic Curve Cryptography
  * RSA
  * XMSS
  * SPHINCS+
  * Dilithium
  * Kyber
* Hybrid Encryption Schemes
* [...]

Along the way, I found two issues:

* Botan's default probability for RSA's `check_key()` is actually _12_ and `strong = true` raises this to _128_. See [here](https://github.com/randombit/botan/blob/3.1.1/src/lib/pubkey/rsa/rsa.cpp#L405). I fixed that along the way.
* `BigInt::random_integer()` has been changed recently and needs adaption in the cryptodoc. See #109. This is not done here.

### CODE REVIEW

**Comment by FAlbertDev** [comment]
Maybe we still want to distinguish between KEM/KEX algorithms and signature algorithms in the new structure. What do you think?

**Comment by reneme** [comment]
> Maybe we still want to distinguish between KEM/KEX algorithms and signature algorithms in the new structure. What do you think?

I would have preferred that, too. Though, it is not really that clear where to put things for ECC, DH/DSA, and in particular, RSA. Therefore, I didn't split it by use case.

**Review by FAlbertDev — APPROVED** [review]
Looks good to me! Only some comments:
- We should remember to consider the restructuring in the changelog
- I think a separation of KEM/KEX and signature algorithms is sensible, in theory. However, I see the problem with RSA and EC crypto; therefore, I think your approach is fine.

File: docs/cryptodoc/src/04_prim_generation.rst [hunk 1]
```diff
@@ -400,6 +400,6 @@ the average case. If the requested probability is higher than 128 the
 algorithm performs more than 50 iterations of the Miller-Rabin primality
 test, which means the generated primes are suitable for usage in
 long-term keys. As the default probability for generating keys is only
-128 in Botan, it is required to call the corresponding ``check_key()``
-functions described in section :ref:`pubkey_key_generation/main` with the ``strong``
+12 in Botan, it is required to call the corresponding ``check_key()``
+functions described in section :ref:`pubkey/rsa` with the ``strong``
 parameter set to true on keys that are intended for long-term usage.
```
[the remark this answers is not in the document]
Comment (reneme) [reply docs/cryptodoc/src/04_prim_generation.rst:405]: True. I read it again, and it should be `128` after all. If `check_key()` is called with `strong = false` the probability for the check is set to 12 only. I got confused by this. Reverting...

**Pull request merged**